spec/models/pipeline_run_spec.rb

```diff
@@ -32,6 +32,14 @@
                                                   **mutates_model_attributes)
         end
 
+        it "contains all dag_jsons for pipeline run stages" do
+          pipeline_run.update_job_status
+
+          pipeline_run.pipeline_run_stages.each do |prs|
+            expect(prs.dag_json).to be_truthy
```

Comment:
Ideally we should check the actual value, but since it might be too hard, I would say we should at least test if this is a valid json.